### PR TITLE
Use asset manifests exclusively to produce/consume previously source built artifacts

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Update the consumption of the previously source-built artifacts and the production of the PackageVersions.props to insert into the bootstrap tarball to both be produced from asset manifests.
 
This allows us to remove the split in the logic between either using the asset manifests or cracking open NuGet packages.

This is a rebased/cleaned up version of https://github.com/dotnet/installer/pull/19585

### CODE REVIEW

File: src/SourceBuild/content/eng/bootstrap/buildBootstrapPreviouslySB.csproj [hunk 1]
```diff
@@ -153,7 +157,7 @@
 
     <!-- Repack tarball with new bootstrap name -->
     <Message Text="  Repacking tarball to $(NewTarballName)" Importance="High" />
-    <Exec Command="tar --numeric-owner -czf $(NewTarballName) *.nupkg *.props SourceBuildReferencePackages/" WorkingDirectory="$(UnpackedTarPath)" />
+    <Exec Command="tar --numeric-owner -czf $(NewTarballName) *.nupkg *.props *.xml SourceBuildReferencePackages/" WorkingDirectory="$(UnpackedTarPath)" />
```
Comment (ViktorHofer) [inline src/SourceBuild/content/eng/bootstrap/buildBootstrapPreviouslySB.csproj:160]: Should the .xml glob also be more specific?

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
The sdk-unified-build test failure is unrelated and will go away when you update your branch to target latest main.

**Comment by ViktorHofer** [comment]
Filed https://github.com/dotnet/source-build/issues/4390 for the sdk-unified-build failure

File: src/SourceBuild/content/eng/bootstrap/buildBootstrapPreviouslySB.csproj (hunk 1 above)
Comment (jkoritzinsky) [reply src/SourceBuild/content/eng/bootstrap/buildBootstrapPreviouslySB.csproj:160]: I wanted to follow the existing pattern, so I did the same thing we do with PackageVersions.props (use a glob). I'd rather leave it that way and clean up both in the future than have one do one thing and another do something different.

**Pull request merged**